### PR TITLE
Rewrite tests with dash testing

### LINKED ISSUE #118: Rewrite the tests
Currently, the tests are using `pytest-dash` and we will need to eventually rewrite the tests using `dash[testing]`. First mentioned here: https://github.com/plotly/dash-cytoscape/pull/117#issuecomment-762871949

### PR DESCRIPTION
**Pull request by kinimesi**
## About
 
Fixes #118, rewrite the tests with `dash[testing]`

## Pre-Merge checklist
- [x] The project was correctly built with `npm run build:all`.
- [ ] If there was any conflict, it was solved correctly.
- [ ] All changes were documented in CHANGELOG.md.
- [x] All tests on CircleCI have passed.
- [ ] All Percy visual changes have been approved.
- [ ] Two people have :dancer:'d the pull request. You can be one of these people if you are a Dash Cytoscape core contributor.

## Reference Issues

Closes #[issue number]

## Other comments

### CODE REVIEW

**Comment by kinimesi** [comment]
I'll look into it, thanks.

**Comment by kinimesi** [comment]
It should be fixed now. However, it needs manual approval. Since `dash_duo.percy_snapshot` combines the snapshot name with the actual python versions, percy cannot match them by name to visually compare (that's why right now it says 103 viaul changes need review)

File: tests/test_callbacks.py [hunk 1]
```diff
@@ -1,101 +1,93 @@
...
+                path = os.path.join(
+                    os.path.dirname(__file__),
+                    'screenshots',
+                    dir_name,
+                    name + '.png'
+                )
+
+                self.dash_duo.driver.save_screenshot(path)
+
+    def test_callbacks(self, dash_duo):
```
Comment (alexcjohnson) [inline tests/test_callbacks.py:77]: Our usual usage of `dash.testing` does not create classes at all - just functions, and they should include test case ids, for which our convention is 4 characters to identify the file and three numbers for the test case within the file. So this one might be `test_cycb001_callbacks(dash_duo)`, in `test_interactions.py` they might be `test_cyin001_dragging`, `test_cyin002_clicking` etc... the advantage here is you can easily run exactly the test you want using for example `pytest -k cyin002`.

Helper functions like `create_input_and_save` can either be nested inside the test case (if they're only used in one place like in this file) or they can be explicitly passed `dash_duo`

File: tests/test_percy_snapshot.py [hunk 2]
```diff
@@ -74,36 +68,20 @@ def display_image(pathname):  # pylint: disable=W0612
 
         return app
 
-    def percy_snapshot(self, name=''):
+    @staticmethod
+    def percy_snapshot(dash_duo, name=''):
         if os.environ.get('PERCY_ENABLED', False):
```
Comment (alexcjohnson) [inline tests/test_percy_snapshot.py:73]: You don't need to explicitly check `PERCY_ENABLED` - `dash_duo.percy_snapshot` is a noop if percy is disabled.

File: tests/test_percy_snapshot.py [hunk 3]
```diff
@@ -116,44 +94,41 @@ def run_percy_on(self, dir_name):
...
 
                 self.percy_snapshot(
-                    name='{}: {}'.format(dir_name.upper(), output_name)
+                    dash_duo, name='{}: {}'.format(dir_name.upper(), output_name)
                 )
                 time.sleep(2)
 
-    def test_usage(self):
-        # Create and start the app
+    def test_usage(self, dash_duo):
```
Comment (alexcjohnson) [inline tests/test_percy_snapshot.py:106]: This whole set of tests would be a great use case for `pytest.mark.parametrize`, eg:
```py
import pytest

@pytest.mark.parametrize('name', ['usage', 'elements', 'layouts', 'style'])
def test_cyps001_snapshots(name, dash_duo):
    app = self.create_app(dir_name=name)
    dash_duo.start_server(app)

    run_percy_on(name, dash_duo)
```

File: tests/test_usage.py [hunk 4]
```diff
@@ -26,48 +20,48 @@ def create_usage_test(self, filename, dir_name='usage'):
...
             os.path.dirname(__file__),
             'screenshots',
             dir_name,
             filename + '.png'
         ))
 
-    def test_usage_advanced(self):
-        self.create_usage_test('usage-advanced')
+    def test_usage_advanced(self, dash_duo):
+        self.create_usage_test(dash_duo, 'usage-advanced')
```
Comment (alexcjohnson) [inline tests/test_usage.py:31]: `pytest.mark.parametrize` again :)

**Comment by kinimesi** [comment]
Thanks @alexcjohnson I mainly rewrote the existing code by using `dash.testing` and kept the existing structure. I'll update them based on your feedback.

**Pull request merged**